src/VirtualClient/VirtualClient.Actions/Wrathmark/WrathmarkWorkloadExecutor.cs

```diff
@@ -0,0 +1,306 @@
+﻿using System.IO;
+
+namespace VirtualClient.Actions.Wrathmark
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    using Microsoft.Extensions.DependencyInjection;
+
+    using VirtualClient.Common;
+    using VirtualClient.Common.Extensions;
+    using VirtualClient.Common.Telemetry;
+    using VirtualClient.Contracts;
+
+    /// <summary>
+    /// Class for the Wrathmark workload.
+    /// </summary>
+    public class WrathmarkWorkloadExecutor : VirtualClientComponent
```

Comment:
Would you be able to write a unit test based on this class or you need help? There are examples in the UNitTests project